src/main/java/net/earthcomputer/clientcommands/task/RenderDistanceScanTask.java

```diff
@@ -17,28 +18,37 @@
 import net.minecraft.world.level.chunk.status.ChunkStatus;
 import net.minecraft.world.phys.Vec3;
 
-import java.util.Iterator;
+import java.util.*;
 import java.util.function.Predicate;
 
 public abstract class RenderDistanceScanTask extends SimpleTask {
     private static final long MAX_SCAN_TIME = 30_000_000L; // 30ms
 
-    private final boolean keepSearching;
+    protected final boolean keepSearching;
 
-    private Iterator<BlockPos.MutableBlockPos> squarePosIterator;
+    private LinkedHashSet<ChunkPos> remainingChunks;
 
     protected RenderDistanceScanTask(boolean keepSearching) {
         this.keepSearching = keepSearching;
     }
 
     @Override
     public void initialize() {
-        squarePosIterator = createIterator();
+        remainingChunks = new LinkedHashSet<>();
+        Entity cameraEntity = Minecraft.getInstance().cameraEntity;
+        if (cameraEntity == null) {
+            _break();
+            return;
+        }
+        BlockPos.spiralAround(new BlockPos(Mth.floor(cameraEntity.getX()) >> 4, 0, Mth.floor(cameraEntity.getZ()) >> 4), Minecraft.getInstance().options.renderDistance().get(), Direction.EAST, Direction.SOUTH).iterator().forEachRemaining(pos -> remainingChunks.add(new ChunkPos(pos.getX(), pos.getZ())));
+        ClientLevelEvents.CHUNK_UPDATE.register((level, pos, oldState, newState) -> remainingChunks.add(new ChunkPos(pos)));
+        ClientLevelEvents.UNLOAD_CHUNK.register((level, pos) -> remainingChunks.remove(pos));
+        ClientLevelEvents.LOAD_CHUNK.register((level, pos) -> remainingChunks.add(pos));
```

Comment:
A few of points here:
- You are registering these listeners without ever unregistering them, leading to a memory leak. Take a look at `ItemThrowTask` for a way to get tasks to listen to events.
- As with `ItemThrowTask`, you probably don't want multiple render distance scan tasks from happening at the same time, so you could add it to the mutex keys (see `ItemThrowTask`).
- These events unconditionally add to the remaining chunks, whereas you probably only want to do that if `keepSearching` is enabled.
- If a block is changed, you may as well check if the changed block matches the block you're looking for (via `scanBlock`), rather than adding the entire chunk to be re-scanned. It does affect the order they're scanned in but I'm not so fussed about that, the main reason it's scanned in this order is to return the found block sooner if it's closer to the player.

Reply:
- I was aware of the memory leak, thought I fixed it
- didn't think of that, thanks
- bruh
- due to the `throws CommandSyntaxException` of `doTick()`, I'd need to handle that in my lambda, what should I do?

Reply:
- I'm not sure what you mean by "bruh". Without `keepSearching` you only want to search once.
- You should try-catch it and report the error to the user, the same as what `onTick()` does.

Reply:
- i didn't notice that i didn't add that
- ok sounds good

Reply:
I can't do what ItemThrowTask does because you can run multiple search tasks at once.

Reply:
That's what the mutex key was for.